airflow/utils/cli.py

```diff
@@ -227,8 +227,12 @@ def setup_locations(process, pid=None, stdout=None, stderr=None, log=None):
         stdout = os.path.join(settings.AIRFLOW_HOME, f'airflow-{process}.out')
     if not log:
         log = os.path.join(settings.AIRFLOW_HOME, f'airflow-{process}.log')
+
     if not pid:
         pid = os.path.join(settings.AIRFLOW_HOME, f'airflow-{process}.pid')
+    else:
+        if not os.path.isabs(pid):
+            raise AirflowException("Path of PID file must be absolute")
```

Comment:
We should make it relative to cwd instead. https://docs.python.org/3/library/os.path.html#os.path.abspath

Reply:
Current working dir is what every unix program treats a relative path as -- we should be no different.

Reply:
That's what I thought of and proposed in the beginning. Please check the discussion at #13200

Reply:
My personal preference aligns with your suggestion.

The current implementation is a "mid-point agreed" after the discussion. Or we should further elaborate the discussion.

Reply:
As far as I see the "https://linux.die.net/man/1/daemonize" commands returns `The 'path' parameter must be an absolute path name` when passed a relative one.  I am not very strong on that, but I think this is expected from a "daemon" program.